### DISCUSSION TITLE
24 ottobre 2022 - esercizio 1

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#23] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/d69c0fa3-5f33-4823-9b53-3ef006108cba)

**Comment by CasuFrost** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/97389762/d338277d-ad02-42e3-a111-fd05297da6ba)

**Comment by Elia-Belli** [comment]
**(1)** 
- Prodotti che costano più di 100 e hanno meno di 50 pezzi in qualche magazzino: $Prod=\pi_{(Codice,Codice-M)}(\sigma_{(N-pezzi\lt 50)}((\sigma_{(Prezzo-unit\gt 100)}Prodotti)\bowtie_{(Codice\ =\ Codice-P)}Scorte))$
> La selezione $Prezzo-unit\gt 100$ è corretta se ogni magazzino ha almeno un pezzo di ogni prodotto, altrimenti i prodotti che non hanno alcun pezzo non compaiono nello schema Scorte
- Risposta: $\pi_{(Codice,Descrizione,Prezzo-unit,Indirizzo)}(Prod \bowtie Prodotti)\bowtie_{}(\rho_{(Codice-M\leftarrow Codice)}Magazzini)$

**(2)**
- Scorte nei magazzini di Roma: $Scorte\\_ RM=\pi_{(Codice-P,Codice-M,N-pezzi)}(\sigma_{(Città\ =\ Roma)}Magazzini)\bowtie_{(Codice\ =\ Codice-M)}Scorte$
- Copia delle Scorte a Roma: $ScorteC = \rho_{(CPc\leftarrow Codice-O, CMc\leftarrow Codice-M,NPc\leftarrow N-pezzi)}Scorte\\_ RM$
- Prodotti che hanno un numero di pezzi minore rispetto ad almeno un altro magazzino: $Prod\\_ Pochi=\pi_{(Codice-P,Codice-M)}(\sigma_{(N-pezzi \lt NPc)}(Scorte\\_ RM\bowtie_{(Codice-P\ =\ CPc)} ScorteC))$
- Magazzini con le scorte maggiori di ciascun prodotto: $Prod\\_ Max= (\pi_{(Codice-P,Codice-M)}Scorte\\_ RM)-Prod\\_ Pochi$
- Risposta: $\pi_{(Codice-P,Descrizione,Codice-M,Indirizzo,N-pezzi)}((Prod\\_ Max \bowtie_{(Codice-P=Codice)} Prodotti)\bowtie_{(Codice-M=Codice)} Magazzini)$
